### PR TITLE
#162414180 controller, model and modelmanager for comment reaction 

### PR DESCRIPTION
**Pull request by tosinibrahim96**
**What does this PR do?**
Creates the model, Controller and model manager for comment reactions

**Description of Task to be completed?**
A user should be able to like a comment
A user should be able to dislike a comment
A user should be able to cancel their reaction 

**How should this be manually tested?**
clone this branch from github and run` npm test`

**What are the relevant pivotal tracker stories?**
[Pivotal Tracker Link](https://www.pivotaltracker.com/story/show/162414180)

### CODE REVIEW

File: constants/index.js [hunk 1]
```diff
@@ -41,15 +42,22 @@ const BAD_REQUEST_CODE = 400,
   UUID = 'ad94832a-19cf-11e9-ab14-d663bd873d45',
   RESET_LINK_SUBJECT = 'Reset Password - Authors Haven',
   BAD_TOKEN = '84ruyhf9q38uyg9',
+  COMMENT_NOT_FOUND = 'Comment does not exist',
+  REACTION_STATUS = 'You have successfully d this comment',
```
Comment (codrex) [inline constants/index.js:63]: `You have successfully d this comment'` typo

File: database/models/articlesComment.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = (sequelize, DataTypes) => {
-  const articlesComment = sequelize.define('articlesComment', {
+  const articlesComments = sequelize.define('articlesComments', {
```
Comment (codrex) [inline database/models/articlesComment.js]: capitalize model names

File: controllers/commentsReactionController.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   *
+   *
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {object| next} response | next function
+   * @memberof commentsReactionController
+   */
+  static async verifyCommentId(req, res, next) {
```
Comment (codrex) [inline controllers/commentsReactionController.js]: This should in the middleware folder

File: lib/modelManagers/commentsReactionModel.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ * comments reaction model manager
+ */
+class commentsReactionModel {
+  /**
+   *
+   *@param {uuid} id The id from request parameter
+   * @memberof commentsReactionModel
+   * @returns {uuid} The parameters from the query
+   */
+  static async confirmCommentId(id) {
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: `static async confirmCommentId(id) ` why is needed?

File: lib/modelManagers/commentsReactionModel.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  /**
+   *
+   *
+   * @static
+   * @param {uuid} commentId
+   * @param {uuid} userId
+   * @returns {Boolean} reactionStatus
+   * @memberof commentsReactionModel
+   */
+  static async checkIfReactionExist(commentId, userId) {
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: This better off as a get reaction function.

File: lib/modelManagers/commentsReactionModel.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+   * @param {string} reaction
+   * @returns {objtec} newReaction
+   * @memberof commentsReactionModel
+   */
+  static async createReaction(commentId, userId, reaction) {
+    if (reaction === 'like') {
+      storedReaction = true;
+    } else {
+      storedReaction = false;
+    }
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: `
if (reaction === 'like') {
      storedReaction = true;
    } else {
      storedReaction = false;
    }
`
?

File: constants/index.js (hunk 1 above)
Comment (tosinibrahim96) [reply constants/index.js:63]: ooh, I am actually inserting a string between the text to format a message later for the user.

File: controllers/commentsReactionController.js (hunk 3 above)
Comment (tosinibrahim96) [reply controllers/commentsReactionController.js]: done

File: lib/modelManagers/commentsReactionModel.js (hunk 4 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: It has been refactored

File: lib/modelManagers/commentsReactionModel.js (hunk 5 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: Done

File: lib/modelManagers/commentsReactionModel.js (hunk 6 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: It has been refactored

File: database/models/articlesComment.js (hunk 2 above)
Comment (tosinibrahim96) [reply database/models/articlesComment.js]: Will be changed

**Review by iAfolayan — DISMISSED** [review]

File: constants/index.js (hunk 1 above)
Comment (codrex) [reply constants/index.js:63]: ok

File: lib/modelManagers/commentsReactionModel.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * comments reaction model manager
+ */
+class commentsReactionModel {
+  /**
+   *
+   *@param {uuid} id The id from request parameter
+   * @memberof commentsReactionModel
+   * @returns {uuid} The parameters from the query
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: ` @returns {uuid} The parameters from the query` this function is returning a boolean due to the fact that you are performing an implicit typecasting here `return !!commentId;`

File: lib/modelManagers/commentsReactionModel.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  static async getComment(id) {
+    const commentId = await ArticlesComments.findOne({ where: { id } });
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: `findOne` in this case will be returning the full record object when the record is found. Why are you then assigning the returned record to a variable with the name `commentId`?

File: lib/modelManagers/commentsReactionModel.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    return !!commentId;
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js]: `return !!commentId;` why are you performing this typecasting?

File: lib/modelManagers/commentsReactionModel.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
+const { ArticlesComments, CommentsReactions } = require('../../database/models');
```
Comment (codrex) [inline lib/modelManagers/commentsReactionModel.js:1]: Why is `ArticlesComments` been imported here?

File: lib/utils/compareReactions.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ *
+ *
+ * @static
+ * @param {string} newReaction
+ * @param {boolean} existingReaction
+ * @returns {boolean} true or false
+ * @memberof commentsReactionModel
+ */
+const compareReactions = (newReaction, existingReaction) => {
```
Comment (codrex) [inline lib/utils/compareReactions.js]: Why is this still here?

File: lib/utils/messageHandler.js [hunk 12]
```diff
@@ -1,16 +1,29 @@
...
+  }),
+  failureResponse = (res, message, statusCode = NOT_FOUND_CODE) => res.status(statusCode).json({
+    success: UNSUCCESSFUL,
+    message
+  }),
+  serverFailure = (res, message, statusCode = SERVER_ERROR_CODE) => res.status(statusCode).json({
+    success: UNSUCCESSFUL,
+    message
+  }),
+  formatReaction = reaction => [REACTION_STATUS.slice(0, 22), reaction, REACTION_STATUS.slice(22)].join('');
```
Comment (codrex) [inline lib/utils/messageHandler.js:23]: `formatReaction ` why is this here?

File: middlewares/valueVerifier.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {object| next} response | next function
+   * @memberof valueVerifier
+   */
+  static async verifyReactionId(req, res, next) {
+    try {
+      const reactionId = await CommentsReactions.findOne({
```
Comment (codrex) [inline middlewares/valueVerifier.js]: Is there no method in the CommentsReactions model manager to get a reaction?

File: lib/modelManagers/commentsReactionModel.js (hunk 7 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: i'll change that now

File: lib/modelManagers/commentsReactionModel.js (hunk 8 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: I'll change that to comment now

File: lib/modelManagers/commentsReactionModel.js (hunk 9 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: I'm trying to check if a comment exist from the controller, this returns true or false. I'lll remove the exclamations

File: lib/modelManagers/commentsReactionModel.js (hunk 10 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js:1]: The article comments model manager has not been merged and I have to check for a valid comment Id before a user can react.

File: lib/utils/compareReactions.js (hunk 11 above)
Comment (tosinibrahim96) [reply lib/utils/compareReactions.js]: This should have been deleted. I'll remove this now

File: lib/utils/messageHandler.js (hunk 12 above)
Comment (tosinibrahim96) [reply lib/utils/messageHandler.js:23]: This is a message handler file and you told me to move `formatreaction` away from the controller. I'm doing all the message formatting here

File: middlewares/valueVerifier.js (hunk 13 above)
Comment (tosinibrahim96) [reply middlewares/valueVerifier.js]: There is one that accepts different parameters. I'll refactor now

File: lib/modelManagers/commentsReactionModel.js (hunk 10 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js:1]: I need the model to do that

File: lib/modelManagers/commentsReactionModel.js (hunk 9 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js]: I just want to know if the comment already exist and i'm not really using the values. Just return true if it exists. That was my thought.

**Review by iAfolayan — APPROVED** [review]

**Review by ibraheemkabir — APPROVED** [review]

File: lib/modelManagers/commentsReactionModel.js (hunk 10 above)
Comment (codrex) [reply lib/modelManagers/commentsReactionModel.js:1]: When` article comments model manager` gets merged please raise a PR to fix this.

`const comment = await ArticlesComments.findOne({ where: { id } });`

**Review by codrex — APPROVED** [review]

File: lib/modelManagers/commentsReactionModel.js (hunk 10 above)
Comment (tosinibrahim96) [reply lib/modelManagers/commentsReactionModel.js:1]: I'll do that
